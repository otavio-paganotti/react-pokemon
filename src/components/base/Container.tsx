import { ReactNode } from "react";

interface IProps {
  children: ReactNode;
}

const container = ({ children }: IProps) => {
  return (
    <>
      <div className="mx-auto max-w-7xl sm:px-6 lg:px-8">
        {children}
      </div>
    </>
  )
}

export default container;