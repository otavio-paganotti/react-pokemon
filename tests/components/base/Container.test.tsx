import React from 'react';
import renderer from 'react-test-renderer';
import Container from '@/components/base/Container';

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
};

test('should render correctly the container component', () => {
  const component = renderer.create(
    <Container>
      This is a content
    </Container>
  );

  let tree = toJson(component);
  expect(tree).toMatchSnapshot();

  expect(tree.props.className).toBe('mx-auto max-w-7xl sm:px-6 lg:px-8');
  expect(tree.children).toHaveLength(1);
  expect(tree.children[0]).toBe('This is a content');
});
