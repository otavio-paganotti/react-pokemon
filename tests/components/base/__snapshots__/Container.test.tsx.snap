// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render correctly the container component 1`] = `
<div
  className="mx-auto max-w-7xl sm:px-6 lg:px-8"
>
  This is a content
</div>
`;
